src/qt/bitcoin.cpp

```diff
@@ -466,6 +468,10 @@ int GuiMain(int argc, char* argv[])
     QApplication::setOrganizationName(QAPP_ORG_NAME);
     QApplication::setOrganizationDomain(QAPP_ORG_DOMAIN);
     QApplication::setApplicationName(QAPP_APP_NAME_DEFAULT);
+    if (gArgs.IsArgSet("-guisettings") && gArgs.GetArg("-guisettings", "") != "") {
```

Comment:
Is it possible to test this logic in UTs?

Reply:
Good question. I'm not an expert with the test framework but we could run `bitcoin-qt --guisettings=/a/custom/path/` and see if `/a/custom/path/Bitcoin/Bitcoin-Qt.conf` exists.
The problem with the Qt tests is that they test the actual GUI like clicking buttons. This belongs more to a startup test, don't know if this exists